src/oscar/templates/oscar/partials/google_analytics_transaction.html

```diff
@@ -1,27 +1,18 @@
 {% extends "partials/google_analytics.html" %}
-{% load l10n %}
+{% load l10n google_analytics_tags %}
 
 {% block extratracking %}
-    ga('require', 'ecommerce', 'ecommerce.js');
-    ga('ecommerce:addTransaction', {
-      'id': '{{ order.number|escapejs }}',
-      'affiliation': '{{ shop_name|escapejs }}',
-      'revenue': '{{ order.total_incl_tax|unlocalize|escapejs }}',
-      'shipping': '{{ order.shipping_incl_tax|unlocalize|escapejs }}',
-      'tax': '{{ order.total_tax|unlocalize|escapejs }}',
-      'currency': '{{ order.currency|escapejs }}'
+    gtag('event', 'purchase', {
+      "transaction_id": '{{ order.number|escapejs }}',
+      "affiliation": '{{ shop_name|escapejs }}',
+      "value": '{{ order.total_incl_tax|unlocalize|escapejs }}',
+      "currency": '{{ order.currency|escapejs }}',
+      "tax": '{{ order.total_tax|unlocalize|escapejs }}',
+      "shipping": '{{ order.shipping_incl_tax|unlocalize|escapejs }}',
+      "items": '{{ order|make_gtag_item_list|escapejs }}'
     });
 
-    {% for line in order.lines.all %}
-    ga('ecommerce:addItem', {
-      'id': '{{ order.number|escapejs }}',
-      'name': '{{ line.title|escapejs }}',
-      'sku': '{{ line.partner_sku|escapejs }}',
-      'category': '{{ line.category|default:"Uncategorised"|escapejs }}',
-      'price': '{{ line.unit_price_incl_tax|unlocalize|escapejs }}',
-      'quantity': '{{ line.quantity|escapejs }}'
+    gtag('event', 'add_to_cart', {
```

Comment:
The user is not adding an item to cart at this stage - this template is rendered on the thank you page after an order is placed. So you need to fire a `purchase` event here. See https://developers.google.com/analytics/devguides/collection/gtagjs/events#recommended-events for what parameters that should contain.